pkg/tinyusb/hw/hw_stm32.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2022 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_tinyusb
+ * @brief
+ * @{
+ *
+ * @brief       tinyUSB hardware driver for STM32 MCUs
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#include <errno.h>
+
+#include "periph_conf.h"
+#include "periph/gpio.h"
+#include "periph/pm.h"
+
+#include "tusb.h"
+#include "device/usbd.h"
+#include "host/usbh.h"
+
+static int tinyusb_hw_init_dev(const stm32_usb_otg_fshs_config_t *conf)
+{
+    /* Block both STOP and STANDBY, TODO STOP is unblocked during USB suspend
+     * status */
+    pm_block(STM32_PM_STOP);
+    pm_block(STM32_PM_STANDBY);
+
+    /* Enable the clock to the peripheral */
+    periph_clk_en(conf->ahb, conf->rcc_mask);
+
+    /* Enables clock on the GPIO bus */
+    gpio_init(conf->dp, GPIO_IN);
+    gpio_init(conf->dm, GPIO_IN);
+    /* Configure AF for the pins */
+    gpio_init_af(conf->dp, conf->af);
+    gpio_init_af(conf->dm, conf->af);
+
+#if 0 /* TODO we don't use USB ID pin for now */
+    gpio_init(conf->id, GPIO_IN);
+#endif
+
+#ifdef USB_OTG_GCCFG_NOVBUSSENS
+#if 0 /* TODO V_USB sensing pin */
+    gpio_init(conf->id, GPIO_OD_PU);
+    gpio_init_af(conf->id, conf->af);
+
+    USB_OTG_GlobalTypeDef *global_regs =
+         (USB_OTG_GlobalTypeDef *)(conf->periph + USB_OTG_GLOBAL_BASE);
+
+    global_regs->GCCFG &= ~USB_OTG_GCCFG_NOVBUSSENS;
+    global_regs->GCCFG |= USB_OTG_GCCFG_VBUSBSEN;
+#else
+    /* Enable no Vbus sensing and enable Power Down Disable */
+    USB_OTG_GlobalTypeDef *global_regs =
+                (USB_OTG_GlobalTypeDef *)(conf->periph + USB_OTG_GLOBAL_BASE);
+
+    /* Enable no Vbus Detect enable  and enable 'Power Down Disable */
+    global_regs->GCCFG |= USB_OTG_GCCFG_NOVBUSSENS | USB_OTG_GCCFG_PWRDWN;
```

Comment:
Ah, I see this is because OTG HS is used on your board. Maybe it is better to use `$(PSRC)/portable/st/synopsys` as driver for STM32. It deals with the specifics in HS mode. In that case, the tinyUSB` variables `CFG_TUSB_RHPORT1_MODE` and `OPT_MODE_HIGH_SPEED` would also have to be set anyway. Unfortunatly, I can't test it.